modules/search/class.jetpack-search-template-tags.php

```diff
@@ -135,6 +153,89 @@ public static function render_filter( $filter, $default_post_types ) {
 		<?php
 	}
 
+	/**
+	 * Renders a single filter for instant search.
+	 *
+	 * @since 8.3.0
+	 *
+	 * @param array $filter             The filter to render.
+	 */
+	public static function render_instant_filter( $filter ) {
+		if ( empty( $filter ) || empty( $filter['buckets'] ) ) {
+			return;
+		}
+
+		$data_base = '';
+		$qv        = $filter['buckets'][0]['query_vars'];
+		$tax_key   = '';
+		switch ( $filter['buckets'][0]['type'] ) {
+			case 'taxonomy':
+				$data_base = 'data-filter-type="' . $filter['buckets'][0]['type'] . '" ';
+				foreach ( $qv as $k => $v ) {
+					$tax_key = $k;
+				}
+				if ( 'category_name' === $tax_key ) {
+					$data_base .= 'data-taxonomy="category"';
+				} elseif ( 'tag' === $tax_key ) {
+					$data_base .= 'data-taxonomy="post_tag"';
+				} else {
+					$data_base .= 'data-taxonomy="' . $tax_key . '"';
+				}
+				break;
+			case 'post_type':
+				$data_base = 'data-filter-type="post_types" ';
+				break;
+			case 'date_histogram':
+				if ( $filter['buckets'][0]['query_vars']['monthnum'] ) {
+					$data_base = 'data-filter-type="month_post_date" ';
+				} else {
+					$data_base = 'data-filter-type="year_post_date" ';
+				}
+				break;
+		}
+
+		?>
+		<h4 class="jetpack-search-filters-widget__sub-heading">
+			<?php echo esc_html( $filter['name'] ); ?>
+		</h4>
+		<ul class="jetpack-search-filters-widget__filter-list">
+			<?php
+			foreach ( $filter['buckets'] as $item ) :
+				$data_str = $data_base . ' ';
+				switch ( $filter['buckets'][0]['type'] ) {
+					case 'taxonomy':
+						$data_str .= 'data-val="' . $item['query_vars'][ $tax_key ] . '"';
+						break;
+					case 'post_type':
+						$data_str .= 'data-val="' . $item['query_vars']['post_type'] . '"';
+						break;
+					case 'date_histogram':
+						if ( $item['query_vars']['monthnum'] ) {
+							$d = sprintf( '%d-%02d-01 00:00:00', $item['query_vars']['year'], $item['query_vars']['monthnum'] );
+						} else {
+							$d = sprintf( '%d-01-01 00:00:00', $item['query_vars']['year'] );
+						}
+						$data_str .= 'data-val="' . $d . '" ';
+						break;
+				}
+				?>
+				<li>
+				<a href="#" class="jetpack-search-filter__link" <?php echo $data_str; ?>>
```

Comment:
$data_str is not fully escaped it needs to be escaped here or be pre-escaped during generation.

Reply:
if pre-escape this means that $data_base also needs escaping (line 182)

Reply:
cc @gibrown

Reply:
added pre-escaping